### PR TITLE
rapidjson: Add version 1.2.0-2022-03-09

### LINKED ISSUE #1924: error: writing 1 byte into a region of size 0
Compiling with gcc with Debug configuration works fine but Release throws the following:

```
/home/david/dev/heartcore/vendor/rapidjson/include/rapidjson/uri.h:330: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
In file included from /home/david/dev/heartcore/vendor/rapidjson/include/rapidjson/pointer.h:19,
                 from /home/david/dev/heartcore/vendor/rapidjson/include/rapidjson/schema.h:19,
                 from /home/david/dev/heartcore/vendor/rapidjson/example/schemavalidator/schemavalidator.cpp:9:
/home/david/dev/heartcore/vendor/rapidjson/include/rapidjson/uri.h: In member function ‘void rapidjson::GenericUri<ValueType, Allocator>::Parse(const Ch*, std::size_t) [with ValueType = rapidjson::GenericValue<rapidjson::UTF8<> >; Allocator = rapidjson::CrtAllocator]’:
/home/david/dev/heartcore/vendor/rapidjson/include/rapidjson/uri.h:330:17: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
  330 |         *query_ = '\0';
      |         ~~~~~~~~^~~~~~
```

### PR DESCRIPTION
**Pull request by haralmha**
Includes PR https://github.com/Tencent/rapidjson/pull/1940 which fixes https://github.com/Tencent/rapidjson/issues/1924 causing errors for newer compilers on centos

### CODE REVIEW

**Comment by haampie** [comment]
can you bump https://github.com/Tencent/rapidjson/issues/1006 again... maybe it's better to move away from rapidjson if they don't do releases.

**Review by tldahlgren — APPROVED** [review]
Confirmed the commit and, from the one for "1.2.0-2021-08-13", that appears to be the way this package is being handled going forward.  

IMO It still feels a bit "off" in at least the fact that this is a merge commit that doesn't even document it as a release.

**Comment by tldahlgren** [comment]
Holding off on the merge to get another opinion on this approach.

**Comment by tldahlgren** [comment]
> can you bump [Tencent/rapidjson#1006](https://github.com/Tencent/rapidjson/issues/1006) again... maybe it's better to move away from rapidjson if they don't do releases.

Not sure how I missed this when I added my comment and approval.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Based on some slack feedback, IF this is the way going forward, @sethrj suggested a different naming scheme for the version (e.g., `2022-03-09` or `master-2022-03-09`) so it is more indicative of what is going on.

In that case, it would be good -- if there are no strong objections -- to also changing the name of the August 13th version as well.

**Comment by haampie** [comment]
I think the [latest known version]-[date] scheme is fine, so that when they finally tag a new release, our versions are ordered correctly.

**Review by haampie — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]

**Comment by tldahlgren** [comment]
> I think the [latest known version]-[date] scheme is fine, so that when they finally tag a new release, our versions are ordered correctly.

I won't stand in the way then.

**Pull request merged**
